lms/djangoapps/cohort_management_dashboard/static/cohort_management_dashboard/js/index.js

```diff
@@ -0,0 +1,119 @@
+                    removeUser(e.target.dataset.username);
+                });
+                $('.member-list tbody .change-cohort-btn').click(function (e) {
+                    moveUser(
+                        e.target.dataset.username,
+                        $(e.target.parentElement).find('select')[0].value
+                    );
+                });
+            }
+
+            function onCohortChanged(newCohortID, page, usersList) {
+                var requestUrl = null;
+                cohortID = newCohortID;
+                if (page === undefined) {
+                    // eslint-disable-next-line no-param-reassign
+                    page = 1;
+                }
+                if (usersList === undefined) {
+                    // eslint-disable-next-line no-param-reassign
+                    usersList = [];
+                }
+
+                requestUrl = options.list_cohort_url.replace(options.template_cohort_id_key, cohortID);
+                requestUrl += '?page=' + page;
+
+                Promise.resolve($.ajax({
+                    url: requestUrl,
+                    type: 'GET'
+                })).then(function (result) {
+                    // eslint-disable-next-line no-param-reassign
+                    usersList = usersList.concat(result.users);
+                    if (result.page === result.num_pages) {
+                        if (usersList.length === 0) {
+                            location.href = options.empty_user_list_redirect;
+                            return;
+                        }
+                        fillTableWithStudent(usersList);
+                    } else {
+                        onCohortChanged(newCohortID, page + 1, usersList);
+                    }
+                });
+            }
+
+            function bindCohortChangeLoster() {
+                $slectControl.change(function (e) {
+                    onCohortChanged(e.target.value);
+                });
+            }
+
+            function displaySelectControl(cohortsList) {
+                function emptyCohortFilter(cohort) {
+                    return cohort.user_count > 0;
+                }
+
+                cohorts = cohortsList.cohorts;
+                $.each(cohorts.filter(emptyCohortFilter), function (key, value) {
+                    $slectControl.append($('<option></option>').attr('value', value.id).text(value.name));
+                });
+                cohortID = options.cohort_id;
+                if (cohortID === undefined || cohortID === null || cohortID === 'None') {
+                    cohortID = cohortsList.cohorts.filter(emptyCohortFilter)[0].id;
+                }
+                bindCohortChangeLoster();
+                $slectControl.find('option[value="' + cohortID + '"]').prop('selected', true);
+                onCohortChanged(cohortID);
+            }
+
+            loadCohortsList().then(displaySelectControl);
```

Comment:
Promise in not looks like required object here, is it correct

Reply:
Refactored